client-js/app/elements/labrad-registry.ts

```diff
@@ -42,15 +43,76 @@ export class LabradRegistry extends polymer.Base {
                                               this.$.editValueInput);
   }
 
+
+  getDefaultSelectedItem(): number {
+    return this.path.length > 0 ? 1 : 0;
+  }
+
+
+  cursorMove(e) {
+    const offset = this.path.length > 0 ? 1 : 0;
+    const length = this.keys.length + this.dirs.length + offset;
+
+    switch (e.detail.combo) {
+      case 'up':
+        if (this.selectedIdx === null || this.selectedIdx === 0) {
+          this.set('selectedIdx', null);
+        } else {
+          this.set('selectedIdx', this.selectedIdx - 1);
+        }
+        break;
```

Comment:
After scrolling up and down for a while on a long registry listing, the selection can go offscreen. It seems we might need to prevent the default event action for up and down keypresses. Of course then we need to implement scrolling in some other way so the listing scrolls to keep the selection on screen; I know there are libraries that can help with this kind of programmatic scrolling control, but I don't know which one is best.